### PR TITLE
Bird!

### PR DESCRIPTION
**Pull request by beck**

### CODE REVIEW

File: lib/feed-bird.js [hunk 1]
```diff
@@ -0,0 +1,4 @@
+module.exports = function(bird, food, cb) {
+  console.log(`Feeding ${bird} to the ${food}...`);
```
Comment (jamesodo) [inline lib/feed-bird.js]: Are you feeding ravens to the corn here?

**Review by nikpl7777 — COMMENTED** [review]
There's a lot of boilprate code. feedBird should be called once. Also, there's callbacks hell, it's better to use Promises/Async functions

File: farm.js [hunk 2]
```diff
@@ -1,18 +1,31 @@
...
 
 
 function main(cb) {
   const cow = new Cow();
   feedCow(cow, function() {
-    eatLunch(function() {
-      cb();
-    });
+    var bird = new Bird();
+    if(bird == 'pigeon') {
```
Comment (nikpl7777) [inline farm.js]: Move feed type to variable and pass it to feedBird directly< without If cascade

File: farm.js [hunk 3]
```diff
@@ -1,18 +1,31 @@
...
+      feedBird(bird, 'seed', function() {
+        eatLunch(function() {
+          cb();
```
Comment (nikpl7777) [inline farm.js]: Make eatLunch returning Promise and use it here

File: lib/feed-bird.js [hunk 4]
```diff
@@ -0,0 +1,4 @@
...
+  setTimeout(cb, 500);
```
Comment (nikpl7777) [inline lib/feed-bird.js:3]: Return here Promise

File: farm.js (hunk 3 above)
Comment (beck) [reply farm.js]: Lets defer the callback > promise refactor and do that after this pull request has landed.

I've opened issue to document that work: https://github.com/yola/farm-foo/issues/2

File: lib/feed-bird.js (hunk 4 above)
Comment (beck) [reply lib/feed-bird.js:3]: Lets introduce promises in the refactor: https://github.com/yola/farm-foo/issues/2

**Review by nikpl7777 — APPROVED** [review]

**Pull request merged**